mlflow/transformers/llm_inference_utils.py

```diff
@@ -73,25 +81,17 @@ def infer_signature_from_llm_inference_task(
     return inferred_signature
 
 
-def convert_data_messages_with_chat_template(data, tokenizer):
+def convert_data_messages_with_chat_template(messages: List[Dict], tokenizer):
     """For the Chat inference task, apply chat template to messages to create prompt."""
-    if "messages" in data.columns:
-        messages = data.pop("messages").tolist()[0]
-    else:
-        raise MlflowException("The 'messages' field is required for the Chat inference task.")
-
     try:
-        messages_str = tokenizer.apply_chat_template(
-            messages, tokenize=False, add_generation_prompt=True
-        )
+        return tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
     except Exception as e:
         raise MlflowException(f"Failed to apply chat template: {e}")
 
-    data["prompt"] = messages_str
-
 
-def preprocess_llm_inference_params(
-    data,
+def preprocess_llm_inference_input(
+    data: pd.DataFrame,
```

Comment:
note: Transformers model saved with `llm/v1/xxx` task always has a signature, so the input is guaranteed to be pandas Dataframe (+ additional guardrail in the function)